src/shogun/classifier/mkl/MKL.cpp

```diff
@@ -241,14 +241,30 @@ class CMKL::Self
 };
 
 CMKL::CMKL(CSVM* s) : CSVM(), svm(NULL), C_mkl(0), mkl_norm(1), ent_lambda(0),
-		mkl_block_norm(1),beta_local(NULL), mkl_iterations(0), mkl_epsilon(1e-5),
+		mkl_block_norm(1),beta_local(NULL),beta_local_size(0), mkl_iterations(0), mkl_epsilon(1e-5),
 		interleaved_optimization(true), w_gap(1.0), rho(0), self()
 {
 	set_constraint_generator(s);
 	self->init();
 
 	SG_DEBUG("creating MKL object %p\n", this)
 	lp_initialized = false ;
+
+	// Register parameters
+	SG_ADD((CSGObject**) &svm, "svm", "wrapper svm", MS_NOT_AVAILABLE);
+	SG_ADD(&C_mkl, "C_mkl", "C mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_norm, "mkl_norm", "norm used in mkl", MS_NOT_AVAILABLE);
+	SG_ADD(&ent_lambda, "ent_lambda", "elastic net sparsity trade-off parameter", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_block_norm, "mkl_block_norm", "mkl sparse trade-off parameter", MS_NOT_AVAILABLE);
+	m_parameters->add_vector(&beta_local, &beta_local_size, "beta_local", "subkernel weights on L1 term of elastic net mkl");
+	SG_ADD(&mkl_iterations, "mkl_iterations", "number of mkl steps", MS_NOT_AVAILABLE);
+	SG_ADD(&mkl_epsilon, "mkl_epsilon", "mkl epsilon", MS_NOT_AVAILABLE);
+	SG_ADD(&interleaved_optimization, "interleaved_optimization", "whether to use mkl wrapper or interleaved opt.", MS_NOT_AVAILABLE);
+	SG_ADD(&w_gap, "w_gap", "gap between interactions", MS_NOT_AVAILABLE);
+	SG_ADD(&rho, "rho", "objective after mkl iterations", MS_NOT_AVAILABLE);
+	SG_ADD(&lp_initialized, "lp_initialized", "if lp is Initialized", MS_NOT_AVAILABLE);
+
+	// Missing: beta_local, W, self, training_time_clock
```

Comment:
Everything that is not registered will not be part of serialization and clone... is it fine with those here?
What about initialization of the memory of them?